src/main/java/com/caucho/hessian/io/SerializerFactory.java

```diff
@@ -487,6 +491,7 @@ public Deserializer getDeserializer(String type)
 
                 deserializer = getDeserializer(cl);
             } catch (Exception e) {
+                _typeNotFoundMap.put(type, NOT_FOUND);
```

Comment:
Cache expiration mechanisms need to be considered, and caching can be a problem here if there is dynamic class loading during application runtime.